cmd/initialize_with_genesis.cpp

```diff
@@ -0,0 +1,258 @@
+
+static void check_rlp_err(rlp::DecodingResult err) {
+    if (err != rlp::DecodingResult::kOk) {
+        throw err;
+    }
+}
+
+static intx::uint256 convert_string_to_uint256(std::string const& str) {
+    intx::uint256 res{0};
+    for(size_t i = 0; i < str.size(); i++) {
+        auto current_digit{str.at(i) - '0'};
+        res += current_digit;
+        res *= 10;
+    }
+    return res/10;
+}
+
+static bool is_hex(std::string const& s) {
+  return s.compare(0, 2, "0x") == 0
+      && s.size() > 2
+      && s.find_first_not_of("0123456789abcdefABCDEF", 2) == std::string::npos;
+}
```

Comment:
I'd invert the order of first two checks.
If size < 2 then compare tries to compare beyond string boundaries